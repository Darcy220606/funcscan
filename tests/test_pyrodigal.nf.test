nextflow_pipeline {

    name "Test pipeline: NFCORE_FUNCSCAN"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_funcscan"
    tag "test_pyrodigal"

    test("test_pyrodigal_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },

                // AMPir
                { assert snapshot("$outputDir/amp/ampir/sample_1_short/sample_1_short.ampir.tsv").match("ampir_tsv_1-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_1_short/sample_1_short.ampir.faa").match("ampir_faa_1-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_short/sample_2_short.ampir.tsv").match("ampir_tsv_2-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_short/sample_2_short.ampir.faa").match("ampir_faa_2-short") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_long/sample_2_long.ampir.tsv").match("ampir_tsv_2-long") },
                { assert snapshot("$outputDir/amp/ampir/sample_2_long/sample_2_long.ampir.faa").match("ampir_faa_2-long") },

                // AMPlify
                { assert snapshot("$outputDir/amp/amplify/sample_1_short/sample_1_short.tsv").match("amplify_tsv_1-short") },
                { assert snapshot("$outputDir/amp/amplify/sample_2_short/sample_2_short.tsv").match("amplify_tsv_2-short") },
                { assert snapshot("$outputDir/amp/amplify/sample_2_long/sample_2_long.tsv").match("amplify_tsv_2-long") },

                // HMMsearch
                { assert snapshot("$outputDir/amp/hmmer_hmmsearch/sample_1_short/sample_1_short_mybacteriocin.txt.gz").match("hmmsearch_output_1-short") },
                { assert snapshot("$outputDir/amp/hmmer_hmmsearch/sample_2_short/sample_2_short_mybacteriocin.txt.gz").match("hmmsearch_output_2-short") },
                { assert snapshot("$outputDir/amp/hmmer_hmmsearch/sample_2_long/sample_2_long_mybacteriocin.txt.gz").match("hmmsearch_output_2-long") },

                // Macrel
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short.smorfs.faa.gz").match("macrel_smorfs_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short.smorfs.faa.gz").match("macrel_smorfs_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long.smorfs.faa.gz").match("macrel_smorfs_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short.all_orfs.faa.gz").match("macrel_all_orfs_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short.all_orfs.faa.gz").match("macrel_all_orfs_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long.all_orfs.faa.gz").match("macrel_all_orfs_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short.prediction.gz").match("macrel_amp_prediction_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short.prediction.gz").match("macrel_amp_prediction_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long.prediction.gz").match("macrel_amp_prediction_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/README.md").match("macrel_readme_file_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/README.md").match("macrel_readme_file_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/README.md").match("macrel_readme_file_2-long") },
                { assert snapshot("$outputDir/amp/macrel/sample_1_short/sample_1_short_log.txt").match("macrel_log_file_1-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_short/sample_2_short_log.txt").match("macrel_log_file_2-short") },
                { assert snapshot("$outputDir/amp/macrel/sample_2_long/sample_2_long_log.txt").match("macrel_log_file_2-long") },

                // TODO: DeepARG

                // TODO: ABRicate

                // TODO: AMRFinderPlus

                // TODO: RGI

                // TODO: fARGene

            )
        }
    }
}
