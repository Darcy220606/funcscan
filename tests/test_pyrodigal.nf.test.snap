{
    "ampir_tsv_2-long": {
        "content": [
            "sample_2_long.ampir.tsv:md5,0a1efb048004b35d93a553a0fe850678"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.536486886"
    },
    "ampir_faa_2-long": {
        "content": [
            "sample_2_long.ampir.faa:md5,4cbce2b01da7ebb91ed194d25209f672"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.550205051"
    },
    "macrel_all_orfs_1-short": {
        "content": [
            "sample_1_short.all_orfs.faa.gz:md5,cb2b6710eefb57e50ceac62d5b11c499"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.655848042"
    },
    "macrel_smorfs_2-short": {
        "content": [
            "sample_2_short.smorfs.faa.gz:md5,24404fcfc2d43eb9ea61982949fa171a"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.632547564"
    },
    "macrel_readme_file_1-short": {
        "content": [
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.733520462"
    },
    "macrel_amp_prediction_2-short": {
        "content": [
            "sample_2_short.prediction.gz:md5,2bed4ad2302371ae2a5802a9292d8a06"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.706524448"
    },
    "hmmsearch_output_2-long": {
        "content": [
            "sample_2_long_mybacteriocin.txt.gz:md5,4d5cf62e2360dc33a2b73692bb2a0079"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.61073358"
    },
    "ampir_faa_1-short": {
        "content": [
            "sample_1_short.ampir.faa:md5,5537c292406bc9edd7c48fb7e89666ca"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.507438087"
    },
    "macrel_log_file_2-short": {
        "content": [
            "sample_2_short_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.786970202"
    },
    "ampir_tsv_2-short": {
        "content": [
            "sample_2_short.ampir.tsv:md5,2fea760bacd9ebc2cb938c728aa4a6b9"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.517645204"
    },
    "hmmsearch_output_1-short": {
        "content": [
            "sample_1_short_mybacteriocin.txt.gz:md5,113ec90adfb019aba9bb8367c359aa45"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.589001598"
    },
    "amplify_tsv_1-short": {
        "content": [
            "sample_1_short.tsv:md5,b8385e808a32a6c97a18ea3d794da72a"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.559556151"
    },
    "amplify_tsv_2-long": {
        "content": [
            "sample_2_long.tsv:md5,73988cf4875db0d245794f47c71bc13d"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.57855296"
    },
    "macrel_amp_prediction_2-long": {
        "content": [
            "sample_2_long.prediction.gz:md5,7167e7c0f269e2ac5935c239c2d13dfe"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.71980231"
    },
    "macrel_readme_file_2-long": {
        "content": [
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.761111667"
    },
    "macrel_smorfs_2-long": {
        "content": [
            "sample_2_long.smorfs.faa.gz:md5,bc48062855750db65cc0f0a14c64e125"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.64404"
    },
    "macrel_readme_file_2-short": {
        "content": [
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.747164318"
    },
    "macrel_smorfs_1-short": {
        "content": [
            "sample_1_short.smorfs.faa.gz:md5,1b5e2434860e635e95324d1804a3be7b"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.621520041"
    },
    "macrel_amp_prediction_1-short": {
        "content": [
            "sample_1_short.prediction.gz:md5,0c4b16e0838be56e012b99169863a168"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.693164084"
    },
    "macrel_all_orfs_2-long": {
        "content": [
            "sample_2_long.all_orfs.faa.gz:md5,0fac84013ec35256a136ca1e0d2cbd48"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.680373948"
    },
    "ampir_tsv_1-short": {
        "content": [
            "sample_1_short.ampir.tsv:md5,e712a287917c3dd17b5f8285d0e97133"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.490101758"
    },
    "macrel_all_orfs_2-short": {
        "content": [
            "sample_2_short.all_orfs.faa.gz:md5,481fb8df3f8ed67010b5152b7fe4059a"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.66783129"
    },
    "amplify_tsv_2-short": {
        "content": [
            "sample_2_short.tsv:md5,a27d9fdef4f13117d16361ea8870e013"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.568720658"
    },
    "macrel_log_file_1-short": {
        "content": [
            "sample_1_short_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.773946713"
    },
    "hmmsearch_output_2-short": {
        "content": [
            "sample_2_short_mybacteriocin.txt.gz:md5,4be5b675e7188d6fb46eae328e452524"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.600155099"
    },
    "macrel_log_file_2-long": {
        "content": [
            "sample_2_long_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.800322405"
    },
    "ampir_faa_2-short": {
        "content": [
            "sample_2_short.ampir.faa:md5,a0a42be0f5aa3b9be5e6482ff88a6fb2"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-07T17:51:04.526000699"
    }
}